.github/workflows/build.yml

```diff
@@ -3,8 +3,17 @@ name: Build
 on:
   push:
     branches: [ main ]
+  paths:
+    - '**'
+    - '!docs/**' # ignore docs changes
+    - '!**.md' # ignore markdown changes
   pull_request:
     branches: [ main ]
+    paths:
+      - '**.go'
+      - 'go.*'
+      - 'cmd/go.*'
+      - 'Makefile'
```

Comment:
This list needs a bit of expansion. My original draft was off the cuff. Take a look through the full file tree for the repo and see if you can spot the ones that are relevant. It might help to filter out the ones this list already matches; that will leave relative few files to consider.
e.g. Some of the make targets depend on `scripts/*.sh` so we need to run tests when [some of?] those change, I think the integration tests depend on the `Dockerfile', etc

Reply:
I have added a few more,
Dockerfile, All scripts, All integration folder files.